### PR TITLE
Merge README and CONTRIBUTING

### PR DESCRIPTION
**Pull request by Ozaq**
CONTRIBUTING.md contained quite a bit of outdated information, most of
this has been removed and the remaining advice has been merged in
README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,84 +4,171 @@
 
 ## Introduction
 
-This repository consists of two modules for simulating and anlyzing pedestrian dynamics. These are:
+This repository consists software for simulating pedestrian dynamics.
```
Comment (chraibi) [inline README.md:7]: of software

File: README.md [hunk 2]
```diff
@@ -4,84 +4,171 @@
...
-2. `jpsreport`: a tool for analyzing the trajectories and measuring the density, flow and velocity.
+Analysis of results can be done with [jpsreport](https://github.com/JuPedSim/jpsreport)
+Visualization of results can be done with [jpsvis](https://github.com/JuPedSim/jpsvis)
 
 ## Building from source
 
-### Linux / macOS
-The build is tested on Ubuntu 18.04 with gcc-9.1 / clang-8 and on macOS Mojave
-10.14.5 with dependencies from Homebrew and Apple LLVM 10.
+It should be possible to build jpscore on all major platforms, however we only
```
Comment (chraibi) [inline README.md:14]: however,

File: README.md [hunk 3]
```diff
@@ -4,84 +4,171 @@
...
+test a few:
 
-#### Requirements
-On Linux you will need a C++17 capable compiler and a standard library that
-supports `<filesystem>`. On macOS you will need the system compiler with C++17
-and filesystem from boost.
+Right now we ensure a working Build for:
```
Comment (chraibi) [inline README.md:17]: now,

File: README.md [hunk 4]
```diff
@@ -4,84 +4,171 @@
...
+* Windows 10
+* MacOS Montery
+* MacOS BigSur
+* Ubuntu 21.10
 
-Recommended:
-* ninja-build
+### Linux / MacOS
+
+We suggested to use the following layout for developing on jpscore:
```
Comment (chraibi) [inline README.md:26]: We suggest using

File: README.md [hunk 5]
```diff
@@ -4,84 +4,171 @@
...
+
+```
+
+#### System Requirements
+
+* C++17 capable compiler
+* clang-format-13
+* wget
+
+Once the above listed requirements are installed build the library dependencies.
```
Comment (chraibi) [inline README.md:42]: above-listed
installed,

File: README.md [hunk 6]
```diff
@@ -4,84 +4,171 @@
...
+
+#### Library Dependencies
+
+##### Linux MacOS
+
+On Linux and MacOS all library dependencies except boost are build from source,
```
Comment (chraibi) [inline README.md:48]: are built

File: README.md [hunk 7]
```diff
@@ -4,84 +4,171 @@
...
+they are eiter part of the source tree and do not need any special attention or
```
Comment (chraibi) [inline README.md:49]: either

File: README.md [hunk 8]
```diff
@@ -4,84 +4,171 @@
...
+they are build with `scripts/setup-deps.sh`. To compile the dependencies invoke
```
Comment (chraibi) [inline README.md:50]: built
the dependencies,

File: README.md [hunk 9]
```diff
@@ -4,84 +4,171 @@
...
+the script from any path where you have write access. The script will create a
+folder called `deps`, download and compile required dependencies. The resulting
+`deps` folder should contain an install tree of all required library
+dependencies. Move / rename this folder to a convenient location, e.g. to
+`jpscore-deps` as outlined in the beginning.
+
+Additionally you will need to install boost >= 1.74 development packages for
```
Comment (chraibi) [inline README.md:57]: Additionally,

File: README.md [hunk 10]
```diff
@@ -4,84 +4,171 @@
...
+your distribution.
+
+##### Windows
+
+On Windows dependencies are installed with `vcpkg`, get it from <vcpkg.io>.
+`vcpkg` will automatically download and install the dependencies listen in
+`vcpkg.json` if used with cmake.
+
+During the CMake invocation listed in the next section add this to your invocation:
```
Comment (chraibi) [inline README.md:66]: section,

File: README.md [hunk 11]
```diff
@@ -4,84 +4,171 @@
...
+```bash
+-DCMAKE_TOOLCHAIN_FILE=<PATH-TO-VCPKG-INSTALLATION>/scripts/buildsystems/vcpkg.cmake
+```
+
+#### How to generate build files with CMake
+
+Now that you have all library dependencies you need to generate build files
```
Comment (chraibi) [inline README.md:73]: dependencies,

File: README.md [hunk 12]
```diff
@@ -4,84 +4,171 @@
...
 cmake -DCMAKE_BUILD_TYPE=Debug <path-to-cmakelists>
 make -j$(nproc)
 ```
 
-The following configuration flags are available:
+### Build Options
 
-##### BUILD_DOC defaults to OFF
-Build internal Doxygen based documentation
+For a list of build options please see [CMakeLists.txt] directly.
```
Comment (chraibi) [inline README.md:96]: options,

File: README.md [hunk 13]
```diff
@@ -4,84 +4,171 @@
...
 
-##### BUILD_WITH_ASAN defaults to OFF (Does not support Windows)
-Build an additional target `jpscore_asan` with address and undefined behavior
-sanitizer enabled. Note there is an approx. 2x slowdown when using
-`jpscore_asan` over `jpscore`
+## Showcase and tutorials
 
-##### CODE_COVERAGE defaults to OFF (Does not support Windows)
-Build unittests with code coverage.
+To highlight some features of JuPedSim we have uploaded some videos on our
```
Comment (chraibi) [inline README.md:104]: JuPedSim,

File: README.md [hunk 14]
```diff
@@ -4,84 +4,171 @@
...
+If you got a question or a problem and need support from our team feel free to
+contact us. You can do this via [email](mailto:dev@jupedsim.org).
 
-## Showcase and tutorials
+Please do not use the issue tracker for personal support requests.
 
-To highlight some features of JuPedSim we have uploaded some videos on our [YouTube channel](https://www.youtube.com/channel/UCKS8w8CUClHEeN4K1SUSMBA).
+### Reporting bugs and requesting features
 
+If you find a bug in the source code, a mistake in the documentation or think
```
Comment (chraibi) [inline README.md:116]: bug,

File: README.md [hunk 15]
```diff
@@ -4,84 +4,171 @@
...
+there is functionality missing you can help us by submitting an issue.
 
-## Support
+If you submit a bug please make sure to include:
+
+* What version / commit id you are using
+* What did you expect the software to do
+* What did the software do instead
+* A description how to reproduce the issue if possible
```
Comment (chraibi) [inline README.md:124]: of how

File: README.md [hunk 16]
```diff
@@ -4,84 +4,171 @@
...
+  details as possible.
+* Provide specific examples to demonstrate the steps.
+* Describe the current behavior and explain which behavior you expected to see
+  instead and why.
+
+## Contributing to JuPedSim
+
+This project is mainly developed by a small group of researchers and students
+from [Jülich Research Center](http://www.fz-juelich.de/en) and
+[BUW](http://www.uni-wuppertal.de/). However you are kindly invited not only to
```
Comment (chraibi) [inline README.md:140]: However,

File: README.md [hunk 17]
```diff
@@ -4,84 +4,171 @@
...
+use JuPedSim but also contributing to our open-source-project. It does not
+matter if you are a researcher, student or just interested in pedestrian
+dynamics. There are only a few rules and advices we want to give to you:
+
+### Workflow
+
+We use a fork based workflow for all development. If you want to contribute any
```
Comment (chraibi) [inline README.md:147]: fork-based

File: README.md [hunk 18]
```diff
@@ -4,84 +4,171 @@
...
+modification we kindly ask you to fork this repository, create a branch with
+your changes on your fork and then open a Pull Request via GitHub.
+
+As part of our CI integration we run formatting checks so make sure your code
+is formatted with clang-format!
+
+If you want to support us by writing the enhancement yourself consider what
```
Comment (chraibi) [inline README.md:154]: yourself,

File: README.md [hunk 19]
```diff
@@ -4,84 +4,171 @@
...
+kind of change it is:
+
+* **Major changes** that you wish to contribute to the project should be
+  discussed first on our **dev mailing list** so that we can better coordinate
+  our efforts, prevent duplication of work, and help you to craft the change so
+  that it is successfully accepted into the project.
+* **Small changes** can be crafted and submitted to our repository as a **pull
+  or merge request**.
+
+Nevertheless open an issue for documentation purposes with the following template:
```
Comment (chraibi) [inline README.md:164]: Nevertheless,

File: README.md [hunk 20]
```diff
@@ -4,84 +4,171 @@
...
+
+### Code formatting
 
-We are heavily working on this project which means that:
+This has become very easy:
 
-- It's not done. We will be releasing new enhancements, bug fixes etc.
-- We love your support. If you find any errors or have suggestions, please write an issue in our [issue-tracker](https://github.com/JuPedSim/jpscore/issues). We will try hard to fix it.
-- Be patient. We are scientists and PhD/master students. Therefore, we primarily care about our research and theses.
+* All code needs to formatted with clang-format
```
Comment (chraibi) [inline README.md:170]: to be

File: README.md [hunk 21]
```diff
@@ -4,84 +4,171 @@
...
+  (A build target for checking / formatting is available: `check-format` /
+  `reformat`)
+* Use only spaces in code
```
Comment (chraibi) [inline README.md:173]: code.

**Review by chraibi — APPROVED** [review]
Kleinigkeiten

File: README.md (hunk 1 above)
Comment (Ozaq) [reply README.md:7]: done

File: README.md (hunk 2 above)
Comment (Ozaq) [reply README.md:14]: done

File: README.md (hunk 3 above)
Comment (Ozaq) [reply README.md:17]: done

File: README.md (hunk 4 above)
Comment (Ozaq) [reply README.md:26]: done

File: README.md (hunk 5 above)
Comment (Ozaq) [reply README.md:42]: done

File: README.md (hunk 6 above)
Comment (Ozaq) [reply README.md:48]: done

File: README.md (hunk 7 above)
Comment (Ozaq) [reply README.md:49]: done

File: README.md (hunk 8 above)
Comment (Ozaq) [reply README.md:50]: done

File: README.md (hunk 9 above)
Comment (Ozaq) [reply README.md:57]: done

File: README.md (hunk 10 above)
Comment (Ozaq) [reply README.md:66]: done

File: README.md (hunk 11 above)
Comment (Ozaq) [reply README.md:73]: done

File: README.md (hunk 12 above)
Comment (Ozaq) [reply README.md:96]: done

File: README.md (hunk 13 above)
Comment (Ozaq) [reply README.md:104]: done

File: README.md (hunk 14 above)
Comment (Ozaq) [reply README.md:116]: I don't think this comma is correct, is it?

File: README.md (hunk 15 above)
Comment (Ozaq) [reply README.md:124]: done

File: README.md (hunk 16 above)
Comment (Ozaq) [reply README.md:140]: done

File: README.md (hunk 18 above)
Comment (Ozaq) [reply README.md:154]: done

File: README.md (hunk 19 above)
Comment (Ozaq) [reply README.md:164]: done

**Pull request merged**